### DISCUSSION TITLE
下载评论区

### DISCUSSION THREAD

**Discussion by Ethkuil** [the1812/Bilibili-Evolved#3708] [功能建议 / Ideas]
将视频下的评论保存为本地文件。我知道这个功能很小众，不过我确实有这个需求……如果作者不方便的话，我看看之后我能否花时间理解一下Bilibili-Evolved的代码然后自行实现。

评论区似乎没有标准的存储格式，这点不同于弹幕，因此可能需要自行定义存储格式。个人认为XML是比较合适的，便于表达回复关系。

麻烦之处可能在于「回复时间」、「点赞数」、「回复人」这些信息到底要不要保留。由我来实现的话，我会将它们都设为可选的。

**Comment by the1812** [comment]
这东西有翻页估计很难存... 翻页的间隔内有人发新评论了就乱掉了
(另外格式建议 JSON

**Reply by stardiviner** [reply]
可以只实现简单的，不考虑复杂情况的。毕竟不是特别专业的需求。不用准确。

主要也是因为手动复制粘贴评论一条一条太麻烦。评论缺少不精准，或者少几页都没啥事。

**Comment by swhoro** [comment]
https://api.bilibili.com/x/v2/reply/main?csrf=###&mode=3&next=2&oid=476763601&plat=1&type=1
这个应该是评论的api，可以试试

**Comment by qtqz** [comment]
其实只是想存下来以后看的话，长截图，或者用 SingleFile 扩展选中所有评论然后保存为 HTML 就好了

存为机器友好的纯文本数据的话（json、md），咱给知乎写过一个类似的，支持展开、翻页评论后增量保存，你可以看看参考写，代码在`parseComments.js`。技术路线为**解析当前网页DOM**，而非请求api。咱暂时没空写BILI的
